ballet/validation/common.py

```diff
@@ -31,21 +93,24 @@ class ChangeCollector:
     """Validate the features introduced in a proposed pull request.
 
     Args:
-        repo (git.Repo): project repo
-        pr_num (int, str): Pull request number
-        contrib_module_path (str): Relative path to contrib module
+        project (ballet.project.Project): project info
+        differ (ballet.util.git.Differ, optional): specific differ to use;
+            if not provided, will be determined automatically from the
+            environment
     """
 
-    def __init__(self, project):
+    def __init__(self, project, differ=None):
         self.project = project
-        self.repo = project.repo
-        self.pr_num = str(project.pr_num)
-        self.contrib_module_path = project.contrib_module_path
+        self.differ = differ
 
-        if can_use_travis_differ():
-            self.differ = TravisPullRequestBuildDiffer(self.pr_num)
-        else:
-            self.differ = LocalPullRequestBuildDiffer(self.pr_num, self.repo)
+        if self.differ is None:
+            if can_use_travis_differ():
+                pr_num = str(self.project.pr_num)
+                self.differ = TravisPullRequestBuildDiffer(pr_num)
+            else:
+                pr_num = str(self.project.pr_num)
```

Comment:
Not directly related to the changes made, but why is `pr_num` a string if the first thing `PullRequestBuildDiffer` does is convert to int?